contrib/inventory/gce_googleapiclient.py

```diff
@@ -186,18 +202,32 @@ def main(args):
     elif all_projects or billing_account_name:
         projects_list = get_all_billing_projects(billing_account_name)
 
+    if zone:
+        zones_list = [zone_name for zone_name in zone]
+
     for project in projects_list:
         try:
-            if zone:
-                zones_list = [zone_name for zone_name in zone]
-            else:
-                zones_list = get_all_zones_in_project(project)
-
-            for zone_name in zones_list:
-                for instance in get_instances(project_id=project,
-                                              zone=zone_name,
-                                              api_version=api_version):
-                    instances.append(instance)
+            if not zone:
+
+                for num in range(MAX_ZONES_THREADS):
+                    thread = Thread(target=get_all_zones_in_project,
+                                    name="ThreadZones_" + str(num),
+                                    args=(project, queue_zones),
+                                    )
+                    thread.start()
+                    thread.join()
```

Comment:
Please create empty list and add threads there. Use another for loop to join the threads on that list.